### PR TITLE
Add `DEPLOYMENT.md` & deployment-related files

### LINKED ISSUE #672: Add `DEPLOYMENT.md` file
## Description

We should add a `DEPLOYMENT.md` file to this repo which describes the various deployment steps that are taken, both manually and automatically. We have some information that's specific to the catalog, since we do automatic deployments of each commit and only perform a deployment via terraform when a new Docker images needs to be shipped.

## Additional context

See the Make WP Slack for more discussion: https://wordpress.slack.com/archives/C02012JB00N/p1660759717725959

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #681: Add deployment-related files to this repo
## Description

We presently have some deployment-related files which are written as part of the EC2 initialization (`init.tpl` file) within the infrastructure repo. We should move these into the catalog repo itself so that it's more clear what is involved in the actual deployment of this service. 

Namely, the `dag-sync.sh` file which pulls new updates from GitHub every few minutes.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #672 by @AetherUnbound
Fixes #681 by @AetherUnbound

## Description

This PR adds a `DEPLOYMENT.md` file (modeled after [an analogous file in the frontend repo](https://github.com/WordPress/openverse-frontend/blob/4703ef7bcda2f7854ce392f9d864f9561a48c303/DEPLOYMENT.md)) which describes the current deployment setup & workflow for both Airflow and Postgres.

I've also moved the `dag-sync.sh` script from the EC2 initialization script in the infrastructure repo into this repo. This makes the deployment process clearer for community contributors and allows us to simplify the EC2 init. I've made some changes to the `dag-sync.sh` script in order to make it more generic and less dependent on our specific Terraform setup.

## Testing Instructions

Please read the documentation and let me know if anything is confusing or if I missed something :slightly_smiling_face:

You can also try running the DAG sync script with `./dag-sync.sh`. Nothing should happen if you just checked out the branch, but if you comment out the following line you should get a warning about a Slack URL not being supplied:

https://github.com/WordPress/openverse-catalog/blob/5fce606112072f39ac0f2693ebf38064b5f9f056/dag-sync.sh#L22

You can add gibberish to the call (e.g. `./dag-sync.sh will-not-work`) and `curl` should throw a fit about not being able to reach `will-not-work`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by AetherUnbound** [comment]
I've added a change to the dag-sync script to ensure that the sync only ever advances by 1 commit at a time. That way our updates to slack show each PR that was merged, rather than potentially skipping several merges to only show the most recent commit. CC @stacimc, just to call out that change 🙂

**Review by stacimc — APPROVED** [review]
This looks so good! The documentation is fantastic, and the update to sync one PR at a time is a really great idea 💯

I thought this would be tricky to test, but it actually worked out quite well -- this branch is not currently rebased on main, which is ahead by 4 commits at the moment. When I initially ran the `dag-sync` it pulled the code at #708, the first of those four 🥳 (And complained about the lack of Slack connection). When I rebased the branch on main, it detected no changes. This looks great!

File: DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Since this process happens automatically, the only precursor to a DAG deployment is merging the relevant PR into `main`.
+
+_**Note**: New DAGs are not automatically enabled and will need to be turned on once deployed._
+
+# Postgres
+
+The Catalog is also backed by a Postgres instance, which serves as the primary data warehouse for all ingested media data.
+
+This instance is hosted on AWS using [RDS](https://aws.amazon.com/rds/).
+The instance can be accessed using [`pgcli`](https://www.pgcli.com/) or a similar client with the jumphost defined in the infrastructure repository.
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply DEPLOYMENT.md:61]: We can! It's in the infrastructure repo, but that should still be okay. I'll add a note that only folks with access will be able to view the link.

File: DEPLOYMENT.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+### Deployment workflow
+
+The "🔒" icon below notes that the action must be performed by a core maintainer with infrastructure access.
+
+1. [Cut a new release on GitHub](https://github.com/WordPress/openverse-catalog/releases).
+2. After the release GitHub action has run, [check that the new Docker tag corresponding to the release exists on ghcr.io](https://github.com/wordpress/openverse-catalog/pkgs/container/openverse-catalog).
+3. Ensure that there are [no actively running DAGs](https://airflow.openverse.engineering/dagrun/list/?_flt_3_state=running).
+4. 🔒 Update the `docker_image_tag` in Terraform's configuration.
+5. 🔒 Initiate the deployment via `just prod apply catalog-airflow`.
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply DEPLOYMENT.md:40]: Ah, good call 🙂

File: DEPLOYMENT.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (stacimc) [reply DEPLOYMENT.md:61]: Good point!

**Review by zackkrida — APPROVED** [review]
The documentation looks great.

**Review by krysal — APPROVED** [review]
These are very detailed instructions ✨ Thanks!
Let's merge it to prevent it from going stale.

**Pull request merged**